minimal_high_density.md

```diff
@@ -0,0 +1,103 @@
+
+This code implements the same model as [`minimal.slim`](minimal.html),
+but implements some "map-based" alternative computational methods so that runtime is linear in density
+instead of quadratic.
+However, these methods are somewhat approximate and may lead to discretization artifacts if density is low,
+so if neighborhood sizes are small (below, say 5 or 10), then the methods should probably not be used.
+
+There are two aspects of the "standard" methods demonstrated in `minimal.slim` that are quadratic in population density
+(i.e., quadratic in the parameter `K`): local density computation, and mate choice.
+The two methods can be used independently - for instance, maybe your simulation
+has small interaction neighborhood size but large mating neighborhood size,
+so you'd use these methods for mating but not density computations.
+
+Both methods are linear because they rely on a pre-computed map of local density.
+The basic computation to do this is
+```
+raw = summarizeIndividuals(p1.individuals, GRID_DIMS, p1.spatialBounds, operation="individuals.size();", perUnitArea=T);
+```
+This makes a grid of dimensions `GRID_DIMS`, counts up the number of individuals in each grid square,
+and divides by the area of the square to get a density per unit area.
+What is an appropriate size for these grid squares?
+Well, we'd like these to be as big as is reasonable,
+because computation will scale with the number of these squares.
+But, this grid is the source of any discretization artifacts:
+the smaller the grid squares are, the closer this model is to `minimal.slim`.
+In both measuring local density and choosing mates we average over a kernel
+with a certain characteristic scale: `SX` for local density (for "interactions") and `SM` for mating.
+So, as long as our grid squares are sufficiently smaller than these,
+any discretization will be smoothed out by those operations.
+So, we set the size of the grid cells to be approximately `min(SX,SM)/2`:
+```
+grid_dims = asInteger(2 * (p1.spatialBounds[c(2,3)] - p1.spatialBounds[c(0,1)]) / min(SX, SM));
+```
+
+The `raw` values are put in two maps: `RAW_DENSITY` and `DENSITY`;
+the first is used for mate choice and the second for interactions.
+
+## Local smoothed density
+
+Local density used as an input for mortality is computed in `minimal.slim`
+by averaging over a Gaussian kernel with standard deviation `SX`.
+To approximate this, we simply smooth the "raw" gridded density by that same kernel:
+```
+DENSITY.smooth(SX * 3, "n", SX);
+```
+and then look up the local, smoothed density at the location of every individual:
+```
+competition = p1.spatialMapValue(DENSITY, inds.spatialPosition);
+```
+This could be made more precise by taking into account that in computing the "raw" density
+also involves some smoothing (the size of the grid squares).
+
```

Comment:
The word "in" seems stray here.  More importantly, I'm not sure what you're suggesting here.  How would you take that fact into account?  What would that look like in script?